tests/basic_relay.go

```diff
@@ -248,15 +242,61 @@ func BasicRelay(network interfaces.LocalNetwork) {
 	// Cancel the command and stop the relayer
 	relayerCancel()
 	_ = relayerCmd.Wait()
+
+	//
+	// Set CatchUpBlockHeight in config
+	//
+	log.Info("Test Setting CatchUpBlockHeight in config")
+
+	// Send three Teleporter messages from subnet A to subnet B
+	log.Info("Sending three Teleporter messages from subnet A to subnet B")
+	_, id1 := sendBasicTeleporterMessage(ctx, subnetAInfo, subnetBInfo, fundedKey, fundedAddress)
+	_, id2 := sendBasicTeleporterMessage(ctx, subnetAInfo, subnetBInfo, fundedKey, fundedAddress)
+	_, id3 := sendBasicTeleporterMessage(ctx, subnetAInfo, subnetBInfo, fundedKey, fundedAddress)
+
+	currHeight, err := subnetAInfo.RPCClient.BlockNumber(ctx)
+	Expect(err).Should(BeNil())
+	log.Info("Current block height", "height", currHeight)
+
+	// Configure the relayer such that it will only process the last of the three messages sent above.
+	// The relayer DB stores the height of the block *before* the first message, so by setting the
+	// CatchUpBlockHeight to the block height of the *third* message, we expect the relayer to skip
+	// the first two messages on startup, but process the third.
+	modifiedRelayerConfig := relayerConfig
+	modifiedRelayerConfig.SourceSubnets[0].CatchUpBlockHeight = currHeight
+	modifiedRelayerConfig.ProcessMissedBlocks = true
+	relayerConfigPath = writeRelayerConfig(modifiedRelayerConfig)
+
+	log.Info("Starting the relayer")
+	relayerCmd, relayerCancel = testUtils.RunRelayerExecutable(ctx, relayerConfigPath)
+	log.Info("Waiting for a new block confirmation on subnet B")
+	<-newHeadsB
+	delivered1, err := subnetBInfo.TeleporterMessenger.MessageReceived(
+		&bind.CallOpts{}, subnetAInfo.BlockchainID, id1,
+	)
+	Expect(err).Should(BeNil())
+	delivered2, err := subnetBInfo.TeleporterMessenger.MessageReceived(
+		&bind.CallOpts{}, subnetAInfo.BlockchainID, id2,
+	)
+	Expect(err).Should(BeNil())
+	delivered3, err := subnetBInfo.TeleporterMessenger.MessageReceived(
+		&bind.CallOpts{}, subnetAInfo.BlockchainID, id3,
+	)
+	Expect(err).Should(BeNil())
+	Expect(delivered1).Should(BeFalse())
+	Expect(delivered2).Should(BeFalse())
+	Expect(delivered3).Should(BeTrue())
+
+	// Cancel the command and stop the relayer
+	relayerCancel()
+	_ = relayerCmd.Wait()
 }
 
-func relayBasicMessage(
-	ctx context.Context,
+func sendBasicTeleporterMessage(ctx context.Context,
 	source interfaces.SubnetTestInfo,
 	destination interfaces.SubnetTestInfo,
 	fundedKey *ecdsa.PrivateKey,
-	fundedAddress common.Address,
-) {
+	fundedAddress common.Address) (teleportermessenger.TeleporterMessage, *big.Int) {
```

Comment:
nit: Can we keep the previous formatting, with the `ctx` on a newline and also the return types on a newline